UIKitTests/UIScrollViewTests.swift

```diff
@@ -47,6 +47,65 @@ class UIScrollViewTests: XCTestCase {
         XCTAssertEqual(scrollView.contentInset, arbitraryContentInset)
     }
 
+    func testScrollIndicatorsViewHierarchyPositions() {
+        // subviews[0] is the back-most view, higher indexes are higher in layer hierarchy
+        // we want scroll indicators to always remain on top of 'normal' subviews
+        // and we want newer 'normal' subviews to be above older ones
+
+        // TODO: iOS does not allow explicit access to scroll indicators via subviews
+        // we should remove that, too - but for now it's the easy way to go and helpful in testing
+
+        let view1 = UIView()
+        let view2 = UIView()
+        let view3 = UIView()
+
+        scrollView.addSubview(view1)
+        scrollView.addSubview(view2)
+        scrollView.addSubview(view3)
+
+        let horizontalIndicatorIndex = scrollView.subviews.index(of: scrollView.horizontalScrollIndicator)!
+        let verticalIndicatorIndex = scrollView.subviews.index(of: scrollView.verticalScrollIndicator)!
+        let label1Index = scrollView.subviews.index(of: view1)!
```

Comment:
I don't think it makes sense to put this into a variable, because (as mentioned below) it will be invalid as soon as we insert another view.

```
XCTAssertGreaterThan(
    scrollView.subviews.index(of: scrollView.horizontalScrollIndicator),
    scrollView.subviews.index(of: view1)
)
```